src/v1/models/ansible.js

```diff
@@ -237,6 +287,10 @@ export default class AnsibleModel extends KubeModel {
     }
     if (response && (response.code || response.message)) {
       throw new Error(`${response.code} - ${response.message}`);
+    } else if (actionType === 'delete') {
+      // if successfully deleting policyAutomation then check & clean up ansible secret
+      // and no need to await cleanAnsibleSecret all done before returning response
+      this.cleanAnsibleSecret({ namespace });
```

Comment:
I'm wondering if there's a way to track the secret that a PolicyAutomation uses and make the cleaning a bit more targeted so that we're not just cleaning when there are no PolicyAutomation objects remaining in the namespace but we're also cleaning when there are no PolicyAutomation objects that use a particular secret. This would also prevent deleting secrets that other non-policy objects might be using (though I'm not sure whether that's relevant).